projects/core/src/product/facade/product-availability.service.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable } from '@angular/core';
+import {
+  Command,
+  CommandService,
+  CommandStrategy,
+} from '../../../src/util/command-query';
+import { Observable } from 'rxjs';
+import { ProductAvailabilities } from '../../model/product.model';
+import { ProductAvailabilityConnector } from '../connectors';
+
+/**
+ * The `ProductAvailabilityService` is responsible for fetching the latest real-time stock
+ * information for products. Unlike the stock information provided by the `ProductService`,
+ * which might be cached, this service ensures that you receive fresh data directly
+ * from the backend.
+ */
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductAvailabilityService {
+  protected connector = inject(ProductAvailabilityConnector);
+  protected command = inject(CommandService);
+  /**
+   * Command to get real-time stock data for a product.
+   */
+  protected getRealTimeStockCommand: Command<
+    { productCode: string; sapCode: string },
+    ProductAvailabilities
+  > = this.command.create(
+    (payload) =>
+      this.connector.getRealTimeStock(payload.productCode, payload.sapCode),
+    {
+      strategy: CommandStrategy.CancelPrevious,
+    }
+  );
+
+  /**
+   * Executes the command to fetch real-time stock data.
+   *
+   * @param productCode The product code for which to fetch stock data.
+   * @param sapCode The SAP code associated with the product.
+   * @returns An observable of `ProductAvailabilities`.
+   */
+  getRealTimeStock(
+    productCode: string,
+    sapCode: string
+  ): Observable<ProductAvailabilities> {
+    return this.getRealTimeStockCommand.execute({ productCode, sapCode });
```

Comment:
[MEDIUM]
`sapCode` is quite a general ambigious term for a newcomer reader of this code. Theoretically it can mean any code that comes from SAP.

To make it more clear, let's rename the arguments of methods `sapCode` -> `unitSapCode`. (but not rename the original backend model properties `product.sapUnit.sapCode`!)

To help with it, I've prepared a proposal sub-PR here. Feel free to review. If you're fine with it, feel free to merge that sub-PR.
https://github.com/SAP/spartacus/pull/19582

Reply:
merged the changes.